### PR TITLE
feat: support integration tests for influxql

### LINKED ISSUE #716: Integration tests for InfluxQL
### Describe This Problem

#698 is supporting InfluxQL, and current implementation has supported some basic query. For the following intensive development on this feature, an integration tests are necessary.

### Proposal

Build the integration tests for InfluxQL similar to the sql integration tests.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by ShiKaiWi**
## Which issue does this PR close?

Closes #716 

## Rationale for this change
Refer to #716  #698

## What changes are included in this PR?
- Support integration tests for InfluxQL;

### CODE REVIEW

File: integration_tests/cases/env/local/influxql/basic.sql [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        (1439827620000, "below 3 feet", "santa_monica", 2.028);
+
+
+-- protocol=influxql
+SELECT * FROM "h2o_feet";
+
+-- protocol=influxql
+SELECT level_description, location, water_level FROM "h2o_feet" where location = 'santa_monica';
+
+DROP TABLE IF EXISTS `h2o_feet`;
```
Comment (Rachelint) [inline integration_tests/cases/env/local/influxql/basic.sql:35]: Newline here.

File: integration_tests/src/database.rs [hunk 2]
```diff
@@ -35,15 +38,71 @@ pub enum DeployMode {
...
     server_process: Option<Child>,
     db_client: Arc<dyn DbClient>,
+    // FIXME: Currently, the new protocol does not support by the dbclient but is exposed by http
+    // service. And remove this client when the new protocol is supported by the dbclient.
+    http_client: Option<HttpClient>,
+}
+
+#[derive(Debug, Clone, Copy)]
+enum Protocol {
+    Sql,
```
Comment (Rachelint) [inline integration_tests/src/database.rs:72]: SQL?

File: Cargo.toml [hunk 3]
```diff
@@ -97,6 +97,7 @@ prost = "0.11"
 query_engine = { path = "query_engine" }
 rand = "0.7"
 remote_engine_client = { path = "remote_engine_client" }
+reqwest = { version = "0.11", features = ["json"] }
```
Comment (jiacai2050) [inline Cargo.toml:100]:
```suggestion
reqwest = { version = "0.11", default-features = false, features = ["rustls-tls", "json"] }
```

File: integration_tests/src/database.rs [hunk 4]
```diff
@@ -35,15 +38,71 @@ pub enum DeployMode {
...
+struct ProtocolParser;
+
+impl ProtocolParser {
+    fn parse_from_ctx(&self, ctx: HashMap<String, String>) -> Protocol {
+        if ctx
+            .get("protocol")
+            .map(|v| v == "influxql")
+            .unwrap_or(false)
+        {
+            Protocol::InfluxQL
```
Comment (jiacai2050) [inline integration_tests/src/database.rs:85]: Add `TryFrom<&str> for Protocol` for this?

**Review by jiacai2050 — APPROVED** [review]
LGTM

File: integration_tests/cases/env/local/influxql/basic.sql [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    VALUES
+        (1439827200000, "between 6 and 9 feet", "coyote_creek", 8.12),
+        (1439827200000, "below 3 feet", "santa_monica", 2.064),
+        (1439827560000, "between 6 and 9 feet", "coyote_creek", 8.005),
+        (1439827560000, "below 3 feet", "santa_monica", 2.116),
+        (1439827620000, "between 6 and 9 feet", "coyote_creek", 7.887),
+        (1439827620000, "below 3 feet", "santa_monica", 2.028);
+
+
+-- protocol=influxql
```
Comment (jiacai2050) [inline integration_tests/cases/env/local/influxql/basic.sql:29]: It should be
```
-- SQLNESS ARG protocol=influxql
```
- https://github.com/CeresDB/sqlness/blob/main/sqlness/examples/interceptor-arg/simple/input.sql#L1

File: integration_tests/cases/env/local/influxql/basic.sql [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+SELECT * FROM "h2o_feet";
+
+-- protocol=influxql
+SELECT level_description, location, water_level FROM "h2o_feet" where location = 'santa_monica';
```
Comment (jiacai2050) [inline integration_tests/cases/env/local/influxql/basic.sql:33]: Add a case for column with double quote
```
SELECT "level_description" FROM "h2o_feet" where location = 'santa_monica';
```

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**